### PR TITLE
small dev doc tweaks

### PR DESCRIPTION
**Pull request by danmoseley**
Some small dev doc tweaks. I'll push up more from time to time as I exercise the docs and trip over something.

cc @wtgodbe

### CODE REVIEW

File: docs/BuildFromSource.md [hunk 1]
```diff
@@ -43,9 +43,11 @@ The steps you follow next depend on your preferred development environment:
     > To modify an existing Visual Studio installation, [follow the instructions for installing from a configuration file](https://learn.microsoft.com/visualstudio/install/import-export-installation-configurations#import-a-configuration) and use the `.vsconfig` file located in the root of the repository:
 
     ```powershell
-    ./eng/scripts/InstallVisualStudio.ps1
+    ./eng/scripts/InstallVisualStudio.ps1 Enterprise
     ```
 
+    Replace `Enterprise` with `Community` if that is your preferred Visual Studio edition. If you use Visual Studio from a preview channel you can specify that also, for example `./eng/scripts/InstallVisualStudio.ps1 Enterprise Preview`.
+
```
Comment (wtgodbe) [inline docs/BuildFromSource.md:50]:
```suggestion
If you are seeing errors similar to `the imported proje
ct "aspnetcore.tools\msbuild\17.1.0\tools\MSBuild\Microsoft\VC\v170\Microsoft.Cpp.Default.prop
s" was not found`, this indicates you need to install/update Visual Studio by running `InstallVisualStudio.ps1`
```

Comment (wtgodbe) [reply docs/BuildFromSource.md:50]: Whoa, the formatting on that got all messed up. But, something similar to this 😆

File: docs/Submodules.md [hunk 2]
```diff
@@ -101,3 +101,9 @@ $GitPromptSettings.EnableFileStatus = $false
 ```
 
 You can disable this permanently by adding to your `$PROFILE` file. (`notepad $PROFILE`)
+
+## Error messages
+
+An error similar to this suggests that the submodules have not be initialized or need to be updated.
```
Comment (ahsonkhan) [inline docs/Submodules.md:107]:
```suggestion
An error similar to this suggests that the submodules have not been initialized or need to be updated.
```

File: docs/BuildFromSource.md [hunk 3]
```diff
@@ -43,9 +43,11 @@ The steps you follow next depend on your preferred development environment:
     > To modify an existing Visual Studio installation, [follow the instructions for installing from a configuration file](https://learn.microsoft.com/visualstudio/install/import-export-installation-configurations#import-a-configuration) and use the `.vsconfig` file located in the root of the repository:
 
     ```powershell
-    ./eng/scripts/InstallVisualStudio.ps1
+    ./eng/scripts/InstallVisualStudio.ps1 Enterprise
     ```
 
+    Replace `Enterprise` with `Community` if that is your preferred Visual Studio edition. If you use Visual Studio from a preview channel you can specify that also, for example `./eng/scripts/InstallVisualStudio.ps1 Enterprise Preview`.
+If you are seeing errors similar to `the imported project "....\aspnetcore.tools\msbuild\17.1.0\tools\MSBuild\Microsoft\VC\v170\Microsoft.Cpp.Default.props" was not found`, this indicates you need to install/update Visual Studio by running `InstallVisualStudio.ps1`
```
Comment (danmoseley) [inline docs/BuildFromSource.md:50]:
```suggestion

    If you are seeing errors similar to `the imported project "....\aspnetcore.tools\msbuild\17.1.0\tools\MSBuild\Microsoft\VC\v170\Microsoft.Cpp.Default.props" was not found`, this indicates you need to install/update Visual Studio by running `InstallVisualStudio.ps1`
```

**Review by wtgodbe — APPROVED** [review]

File: docs/Submodules.md [hunk 4]
```diff
@@ -101,3 +101,9 @@ $GitPromptSettings.EnableFileStatus = $false
 ```
 
 You can disable this permanently by adding to your `$PROFILE` file. (`notepad $PROFILE`)
+
+## Error messages
```
Comment (captainsafia) [inline docs/Submodules.md:105]: We can include a new section in https://github.com/dotnet/aspnetcore/blob/main/docs/BuildErrors.md#build-errors and clarify how to resolve it by running `git submodule update --init --recursive`

**Pull request merged**